components/tooltip/follow-scroll.directive.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @license
+ * Copyright Alibaba.com All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { Directive, HostListener } from '@angular/core';
+import { Subject } from 'rxjs';
+
+@Directive({
+  selector: '[nz-follow-scroll]'
+})
+export class NzFollowScrollDirective {
+  scroll$ = new Subject<void>();
+
+  @HostListener('scroll')
```

Comment:
Thanks for your pull request!
some comments here:
1. do not use `HostListener('scroll')`, it will cause the performance problem with zone.js
2. if you want to define scroll strategy, have a look at https://material.angular.io/cdk/scrolling/overview
3. issue is needed before creating any pull request to help others understand the proposal.

Reply:
I read the website what u provide. And I find we just need to add `cdkScrollable` directive rather than change tooltip's code. So I revert the commit about the directive `nz-follow-scroll` I made, just update the docs.